theia-extensions/viewer-prototype/src/node/trace-server-service.ts

```diff
@@ -11,25 +11,33 @@ export class TraceServerServiceImpl implements TraceServerConfigService {
     private server?: ChildProcess;
     private port?: number;
 
-    async startTraceServer(path?: string, port?: number): Promise<string> {
+    async startTraceServer(path?: string, cliArgs?: string): Promise<string> {
         if (path === undefined) {
             throw new Error('no Trace Server path specified');
-        } else if (port === undefined) {
+        } else if (cliArgs === undefined) {
+            throw new Error('no Trace Server cli arguments specified');
+        }
+
+        const port = cliArgs.match(/(?<=port=)\d+/g);
```

Comment:
The port doesn't need to be parsed because it's Trace Compass server specific and this patch aims the server to be independent of the server implementation specifics.